controllers/k8ssandra/cleanup.go

```diff
@@ -310,16 +310,14 @@ func (r *K8ssandraClusterReconciler) findDcForDeletion(
 			}
 		}
 	} else {
-		for _, remoteClient := range r.ClientCache.GetRemoteClients() {
-			err := remoteClient.List(ctx, dcList, options)
-			if err != nil {
-				return nil, nil, fmt.Errorf("failed to find CassandraDatacenter (%s) for deletion: %v", dcName, err)
-			}
+		err := remoteClient.List(ctx, dcList, options)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to find CassandraDatacenter (%s) for deletion: %v", dcName, err)
+		}
```

Comment:
Not directly related to the fix, but I think this was a typo. See `findStargateForDeletion` and `findReaperForDeletion`.